e2e/suites/provider/cases/azure/provider.go

```diff
@@ -47,23 +48,26 @@ type azureProvider struct {
 func newazureProvider(f *framework.Framework, clientID, clientSecret, tenantID, vaultURL string) *azureProvider {
 	clientCredentialsConfig := kvauth.NewClientCredentialsConfig(clientID, clientSecret, tenantID)
 	clientCredentialsConfig.Resource = "https://vault.azure.net"
-	authorizer, err := clientCredentialsConfig.Authorizer()
```

Comment:
I really wanted to change the azure provider initialization because if you do not specify `AZURE_*` env vars it will fail.

With this we move the initialization to a `BeforeEach()` block where it is called exactly once to do the initialization only if the test cases are selected.